addons/frag/functions/fnc_doFragTargeted.sqf

```diff
@@ -0,0 +1,176 @@
+#include "..\script_component.hpp"
+/*
+ * Author: Jaynus, NouberNou, Lambda.Tiger
+ * This function creates fragments targeted at specific entities, up to
+ * a configured maximum.
+ *
+ * Arguments:
+ * 0: Position of fragmenting projectile ASL <ARRAY>
+ * 1: Velocity of the fragmenting projectile <ARRAY>
+ * 2: Maximum range of fragments to calculate <NUMBER>
+ * 3: Maximum number of fragments to produce <NUMBER>
+ * 4: Types of fragments <ARRAY>
+ * 5: A modified parameter used to calculate whether a fragment hits <NUMBER>
+ * 6: Shot parent <ARRAY>
+ *
+ * Return Value:
+ * Number of fragments created <NUMBER>
+ *
+ * Example:
+ * [getPosASL _proj, velocity _proj, 50, 50, [], 1, [player, player]] call ace_frag_fnc_doFragTargeted;
+ *
+ * Public: No
+ */
+
+params [
+    "_posASL",
+    "_fragVelocity",
+    "_fragRange",
+    "_maxFrags",
+    "_fragTypes",
+    "_modFragCount",
+    "_shotParents"
+];
+TRACE_5("fnc_doFragTargeted",_posASL,_fragRange,_maxFrags,_fragTypes,_modFragCount);
+
+if (_fragTypes isEqualTo []) then {
+    _fragTypes = [
+        QGVAR(tiny), QGVAR(tiny), QGVAR(tiny),
+        QGVAR(tiny_HD), QGVAR(tiny_HD), QGVAR(tiny_HD),
+        QGVAR(small), QGVAR(small), QGVAR(small), QGVAR(small),
+        QGVAR(small_HD), QGVAR(small_HD), QGVAR(small_HD), QGVAR(small_HD),
+        QGVAR(medium_HD), QGVAR(medium_HD), QGVAR(medium_HD), QGVAR(medium_HD), QGVAR(medium_HD)
+    ];
+};
+
+// Post 2.18 change - uncomment line 47, remove line 49, 56-62, 70-72, and change lines 64 & 170 from _targetse to _objects
+// private _objects = [ASLToAGL _posASL, _fragRange, _fragRange, 0, false, _fragRange ] nearEntities [["Car", "Motorcycle", "Tank", "StaticWeapon", "CAManBase", "Air", "Ship"], false, true, true];
+
+private _objects = (ASLToATL _posASL) nearEntities [["Car", "Motorcycle", "Tank", "StaticWeapon", "CAManBase", "Air", "Ship"], _fragRange];
+if (_objects isEqualTo []) exitWith {
+    TRACE_2("No nearby targets",_posASL,_fragRange);
+    0;
+};
+
+// grab crews and add them in so that targets stay approx. sorted by distance
+private _targets = [];
+{
+    private _crew = crew _x;
+    _crew pushBackUnique _x;
+    _targets append _crew;
+
+} forEach _objects;
+
+TRACE_3("Targets found",_posASL,_fragRange,count _targets);
+
+// limit number of fragments per direction (2D) to 10 using _fragArcs
+private _fragArcs = createHashMap;
+private _totalFragCount = 0;
+{ // Begin of forEach iterating on _targets
+    if (!alive _x) then {
```

Comment:
It's a hold over from frago that I think fits. The current setup limits fragment count to 50 because of the performance cost of targeting and creating that many fragments. I think it's preferential to use those fragments on alive units. We could add a method of keeping track of dead entities and itterate through that once we clear through all the important targets, but I don't see the point of targeting dead entities.

Reply:
Any thoughts on this? Otherwise, I'll leave it as is.

Reply:
Maybe take the performance hit and change line 50 to
```sqf
private _targets = [];
private _deadTargets = [];
{
    private _crew = crew _x;
    _crew pushBackUnique _x;
    _crew apply {
        if (alive _x) then {
            _targets pushBack _x;
        } else {
            _deadTargets pushBack _x;
        };
    };
} forEach _objects;

_targets append _deadTargets;
```